flow-client/src/main/frontend/form/Validators.ts

```diff
@@ -97,8 +97,8 @@ export class Email extends AbstractValidator<string> {
   constructor(attrs?: ValidatorAttributes) {
     super({ message: 'must be a well-formed email address', ...attrs });
   }
-  validate(value: string | undefined) {
-    return value === undefined || isEmail(value);
+  validate(value: string | null | undefined) {
+    return !value || isEmail(value);
```

Comment:
It also rejects the empty string. Is it ok? If not, we can replace it with the loose null check:
```suggestion
    return value != null || isEmail(value);
```

Reply:
I also thought about this. I think the Email validator should accept the empty string, which could happen when the user first type sth, then remove the value. 
This kind of required check should be done in the Required/NotNull validator, not this Email validator.

Reply:
Do I understand correctly that results of `!value` and `value != null` are the same then?

Reply:
No. The difference is `''`. `''` passes the 'Email' validation with `!value`, fails with `value !=null`. As indicated in the changes of another file.
```
const validator = new Email();
assert.isTrue(validator.validate(''));
```